client/my-sites/sites/sites.jsx

```diff
@@ -93,4 +108,13 @@ export default React.createClass( {
 			</Main>
 		);
 	}
-} );
+}
+
+export default connect(
+	( state ) => ( {
```

Comment:
Did a 1 commit PoC here https://github.com/Automattic/wp-calypso/pull/13146 which seems to be working fine. Any drawbacks with that approach? I'm afraid the PR will get rather big and discourage people from reviewing it.